decentraland/screens/MyAssetsScreen.js

```diff
@@ -1,42 +1,76 @@
 import React from "react";
 import { connect } from "react-redux";
-import { removeFromMyAssetsList, appendToMyAssetsList } from "../redux/actions";
+import {
+  removeFromMyAssetsList,
+  appendToMyAssetsList,
+  addUserAction,
+} from "../redux/actions";
 import PropTypes from "prop-types";
 import MyAssetsList from "@presentational/MyAssetsList";
 import MyAssetsListItem from "@presentational/MyAssetsListItem";
 import { listsAreEqual, getContracts, logWarn } from "@helpers";
 import { generateAssetFromId } from "@helpers/decentraland";
 import DecentralandUtils from "tasit-sdk/dist/helpers/DecentralandUtils";
-import MyAssetStatus from "@constants/MyAssetStatus";
-const { BOUGHT } = MyAssetStatus;
+import { SUCCESSFUL } from "@constants/UserActionStatus";
 
 export class MyAssetsScreen extends React.Component {
   componentDidMount = async () => {
     const {
       account,
-      myAssets: assetsFromState,
       removeFromMyAssetsList,
       appendToMyAssetsList,
+      addUserAction,
     } = this.props;
     if (account) {
       const { address } = account;
 
-      // Note: If a stored asset doesn't have a status, assuming 'bought'.
-      const boughtAssets = assetsFromState
-        .map(asset => (!asset.status ? { ...asset, status: BOUGHT } : asset))
-        .filter(asset => asset.status === BOUGHT);
+      const boughtAssets = this._getBoughtAssetsFromState();
+
+      const {
+        assetsFromBlockchain,
+        actionsFromBlockchain,
+      } = await this.__getAssetsAndActionsFromBlockchain(address);
 
-      const assetsFromBlockchain = await this._getAssetsFromBlockchain(address);
       const shouldUpdate = !listsAreEqual(assetsFromBlockchain, boughtAssets);
       if (shouldUpdate) {
         // TODO: Add some UI indication that something unexpected happened
         this._logAssetsInconsistency(assetsFromBlockchain, boughtAssets);
+
         removeFromMyAssetsList(boughtAssets);
         appendToMyAssetsList(assetsFromBlockchain);
+        addUserAction(actionsFromBlockchain);
       }
     }
   };
 
+  _getAssetsAndActionsFromBlockchain = async address => {
+    const assetsFromBlockchain = await this._getAssetsFromBlockchain(address);
+
+    const actionsFromBlockchain = assetsFromBlockchain.map(asset => {
+      const { actionId, id: assetId } = asset;
+      const status = SUCCESSFUL;
+      return { actionId, assetId, status };
+    });
+
+    assetsFromBlockchain.forEach(asset => delete asset.actionId);
```

Comment:
Is this temporary, or will we always need to do this?

Reply:
(Not gating merging this PR)

Reply:
Since `sellOrders` from `DecentralandUtils` have `transactionHash`/`actionId` and we want to keep this info only on `userActions` state, I think it could be needed.

Reply:
Thanks for the explanation. Is that the most compact/convenient way for the utils to return this information given how we'll use it going forward?

Reply:
I think so,
The source of that info is `CreatedOrder` events from Marketplace contract. So, each event has the assetId, type (nftAddress) and actionId (txhash).

Reply:
Ah right, so there's not a separate lookup. Okay, makes sense! Thanks for the explanation. Resolving this.